test/permissionsTest.js

```diff
@@ -0,0 +1,44 @@
+const { it, describe } = require('mocha');
+const assert = require('assert');
+const {
+    validArrayInputs, invalidArrayInputs, expectedHexResult, expectedArrayResult, invalidHexInputs,
+} = require('./permissionsMockData');
+const { encodeData, decodeData } = require('../src/models/permissions/permissionHelper');
+
+describe('Encoding functionality', () => {
+    describe('Testing encoding function', () => {
+        it('Should encode to hex', async () => {
+
+            validArrayInputs.forEach((item, index) => {
+                const response = encodeData(item);
+                assert.deepStrictEqual(response, expectedHexResult[index]);
+            });
+        });
+        it('Should fail on invalid input', async () => {
+            invalidArrayInputs.forEach((item) => {
+                assert.throws(() => encodeData(item), {
+                    message: 'Invalid input',
+                    name: 'Error',
+                });
+            });
+        });
+    });
+
+    describe('Testing decoding function', () => {
```

Comment:
We should have test that proves it works fine for mixed lower/uppercase in hexadecimal